src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CompoundAssignmentOperator.cs

```diff
@@ -491,6 +491,7 @@ private BoundExpression TransformCompoundAssignmentLHS(BoundExpression originalL
             // we have a variable on the left. Transform it into:
             // ref temp = ref variable
             // temp = temp + value
+            var localKind = needsSpilling ? SynthesizedLocalKind.Spill : SynthesizedLocalKind.LoweringTemp;
```

Comment:
Seems worrying if we create a ref temp that needs spilling since that's not possible.

Reply:
Turns out this is an existing problem with compound assignment in general. Filed [https://github.com/dotnet/roslyn/issues/36443](https://github.com/dotnet/roslyn/issues/36443)

---
In reply to: [293113591](https://github.com/dotnet/roslyn/pull/36329#discussion_r293113591) [](ancestors = 293113591)